libraries/rush-lib/src/api/LastInstallFlag.ts

```diff
@@ -80,6 +82,29 @@ export class LastInstallFlag {
                 `New Path: ${newStorePath}`
             );
```

Comment:
I was surprised to find that prior to this PR, Rush deletes **last-install.flag** flag when doing a subset install.

For example, if I run `rush install` twice in a row, the second time it prints `Installation is already up-to-date.` and skips invoking PNPM entirely.  Whereas (prior to this PR) if I run `rush install --to library` twice in a row, it would invoke PNPM both times, because **last-install.flag** flag is not created with `--to`.

This PR improves the behavior: If I run `rush install --to library` twice, the second time it does print `Installation is already up-to-date.` and skip `pnpm install` (although it runs `pnpm rebuild --pending` both times, by design).  That seems like an improvement. 👍

@D4N14L @iclanton Do you remember why this optimization wasn't implemented previously?  Is adding `"selectedProjectNames"` to **last-install.flag** flag sufficient to guarantee a valid install?  Are there any interesting edge cases we should test?

Reply:
As long as the options to the previous install are an exact match, it should be fine to allow reuse. Adding or removing projects should cause a re-install, though.

Reply:
The reason _removing_ projects should cause a reinstall is because, in the presence of hoisting (i.e. unless hoist=false in .npmrc), a filtered install and a full install will have different packages visible to the dependencies of the selected projects. If we want to enforce use of `--purge` to reproduce the behavior of a clean install, then we could potentially allow subset installs to be a no-op.